### PR TITLE
[CI fest] Try to fix `test_broken_projections/test.py::test_broken_ignored_replicated`

### LINKED ISSUE #66835: Test `test_broken_projections/test.py::test_broken_ignored_replicated` is flaky
**Describe the bug**

https://s3.amazonaws.com/clickhouse-test-reports/0/2beed8c30413cd091c774c6246acfdad507f5d27/integration_tests__release__[1_4].html

**Additional information**

Previous attempt to fix this test was here: https://github.com/ClickHouse/ClickHouse/pull/61932
According to the CI database this test failed `4 times` in the past 30 days, [the report itself is here. ](https://play.clickhouse.com/play?user=play#U0VMRUNUIGNoZWNrX25hbWUsIHRlc3RfbmFtZSwgcmVwb3J0X3VybApGUk9NIGNoZWNrcwpXSEVSRSAxCiAgICBBTkQgY2hlY2tfc3RhcnRfdGltZSA+PSBub3coKSAtIElOVEVSVkFMIDMwIERBWVMKICAgIEFORCBwdWxsX3JlcXVlc3RfbnVtYmVyID0gMAogICAgQU5EIHRlc3Rfc3RhdHVzICE9ICdTS0lQUEVEJwogICAgQU5EIHRlc3Rfc3RhdHVzIExJS0UgJ0YlJwogICAgQU5EIGNoZWNrX3N0YXR1cyAhPSAnc3VjY2VzcycKICAgIEFORCBjaGVja19uYW1lIE5PVCBMSUtFICdTUUxhbmNlciUnCiAgICBBTkQgdGVzdF9uYW1lIElMSUtFICcldGVzdF9icm9rZW5faWdub3JlZF9yZXBsaWNhdGVkJScKT1JERVIgQlkgY2hlY2tfbmFtZSwgdGVzdF9uYW1lLCBjaGVja19zdGFydF90aW1l)

**Error message and/or stacktrace**

```
=================================== FAILURES ===================================
________________________ test_broken_ignored_replicated ________________________
[gw2] linux -- Python 3.10.12 /usr/bin/python3

cluster = <helpers.cluster.ClickHouseCluster object at 0x7f883ae82950>

    def test_broken_ignored_replicated(cluster):
        node = cluster.instances["node"]
    
        table_name = "test3"
        table_name2 = "test3_replica"
        create_table(node, table_name, 1)
    
        insert(node, table_name, 0, 5)
        insert(node, table_name, 5, 5)
        insert(node, table_name, 10, 5)
        insert(node, table_name, 15, 5)
    
        check(node, table_name, 1)
    
        create_table(node, table_name2, 2, table_name)
>       check(node, table_name2, 1)

test_broken_projections/test.py:436: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

node = <helpers.cluster.ClickHouseInstance object at 0x7f883aee40d0>
table = 'test3_replica', check_result = 1, expect_broken_part = ''
expected_error = '', do_check_command = True

    def check(
        node,
        table,
        check_result,
        expect_broken_part="",
        expected_error="",
        do_check_command=True,
    ):
        if expect_broken_part == "proj1":
            assert expected_error in node.query_and_get_error(
                f"SELECT c FROM '{table}' WHERE d == 12 ORDER BY c SETTINGS force_optimize_projection_name = 'proj1'"
            )
        else:
            query_id = node.query(
                f"SELECT queryID() FROM (SELECT c FROM '{table}' WHERE d == 12 ORDER BY c SETTINGS force_optimize_projection_name = 'proj1')"
            ).strip()
            for _ in range(10):
                node.query("SYSTEM FLUSH LOGS")
                res = node.query(
                    f"""
                SELECT query, splitByChar('.', arrayJoin(projections))[-1]
                FROM system.query_log
                WHERE query_id='{query_id}' AND type='QueryFinish'
                """
                )
                if res != "":
                    break
            if res == "":
                res = node.query(
                    """
                    SELECT query_id, query, splitByChar('.', arrayJoin(projections))[-1]
                    FROM system.query_log ORDER BY query_start_time_microseconds DESC
                """
                )
                print(f"Looked for query id {query_id}, but to no avail: {res}")
>               assert False
E               assert False
```

### PR DESCRIPTION
**Pull request by zvonand**
Closes #66835

### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

#### CI Settings (Only check the boxes if you know what you are doing):
- [ ]  Allow: All Required Checks
- [ ]  Allow: Stateless tests
- [ ]  Allow: Stateful tests
- [x]  Allow: Integration Tests
- [ ]  Allow: Performance tests
- [ ]  Allow: All Builds
- [ ]  Allow: batch 1, 2 for multi-batch jobs
- [ ]  Allow: batch 3, 4, 5, 6 for multi-batch jobs
---
- [ ]  Exclude: Style check
- [ ]  Exclude: Fast test
- [ ]  Exclude: All with ASAN
- [ ]  Exclude: All with TSAN, MSAN, UBSAN, Coverage
- [ ]  Exclude: All with aarch64, release, debug
---
- [ ]  Do not test
- [ ]  Woolen Wolfdog
- [ ]  Upload binaries for special builds
- [ ]  Disable merge-commit
- [ ]  Disable CI cache

### CODE REVIEW

**Comment by robot-clickhouse-ci-2** [comment]
*This is an automated comment for commit 8fb560d2575c121b252ab1e6d8e13f9486dc2b38 with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/66915/8fb560d2575c121b252ab1e6d8e13f9486dc2b38/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Builds</td><td>There's no description for the check yet, please add it to tests/ci/ci_config.py:CHECK_DESCRIPTIONS</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/bd9bfc34a932b732e08860abbe76dac10f9ee46b/builds/report.html">✅ success</a></td></tr>
<tr><td>ClickBench</td><td>Runs [ClickBench](https://github.com/ClickHouse/ClickBench/) with instant-attach table</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/bd9bfc34a932b732e08860abbe76dac10f9ee46b/clickbench__aarch64_.html">✅ success</a></td></tr>
<tr><td>Compatibility check</td><td>Checks that <b>clickhouse</b> binary runs on distributions with old libc versions. If it fails, ask a maintainer for help</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/bd9bfc34a932b732e08860abbe76dac10f9ee46b/compatibility_check__aarch64_.html">✅ success</a></td></tr>
<tr><td>Docker keeper image</td><td>The check to build and optionally push the mentioned image to docker hub</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/bd9bfc34a932b732e08860abbe76dac10f9ee46b/docker_keeper_image.html">✅ success</a></td></tr>
<tr><td>Docker server image</td><td>The check to build and optionally push the mentioned image to docker hub</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/bd9bfc34a932b732e08860abbe76dac10f9ee46b/docker_server_image.html">✅ success</a></td></tr>
<tr><td>Docs check</td><td>Builds and tests the documentation</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/91693daf599bbaeaed5adb8cba257c176628e2be/docs_check.html">✅ success</a></td></tr>
<tr><td>Fast test</td><td>Normally this is the first check that is ran for a PR. It builds ClickHouse and runs most of <a href="https://clickhouse.com/docs/en/development/tests#functional-tests">stateless functional tests</a>, omitting some. If it fails, further checks are not started until it is fixed. Look at the report to see which tests fail, then reproduce the failure locally as described <a href="https://clickhouse.com/docs/en/development/tests#functional-test-locally">here</a></td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/bd9bfc34a932b732e08860abbe76dac10f9ee46b/fast_test.html">✅ success</a></td></tr>
<tr><td>Install packages</td><td>Checks that the built packages are installable in a clear environment</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/bd9bfc34a932b732e08860abbe76dac10f9ee46b/install_packages__aarch64_.html">✅ success</a></td></tr>
<tr><td>Performance Comparison</td><td>Measure changes in query performance. The performance test report is described in detail <a href="https://github.com/ClickHouse/ClickHouse/tree/master/docker/test/performance-comparison#how-to-read-the-report">here</a>. In square brackets are the optional part/total tests</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/bd9bfc34a932b732e08860abbe76dac10f9ee46b/performance_comparison__release__[1_4]/report.html">✅ success</a></td></tr>
<tr><td>Stateful tests</td><td>Runs stateful functional tests for ClickHouse binaries built in various configurations -- release, debug, with sanitizers, etc</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/bd9bfc34a932b732e08860abbe76dac10f9ee46b/stateful_tests__aarch64_.html">✅ success</a></td></tr>
<tr><td>Stateless tests</td><td>Runs stateless functional tests for ClickHouse binaries built in various configurations -- release, debug, with sanitizers, etc</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/bd9bfc34a932b732e08860abbe76dac10f9ee46b/stateless_tests__aarch64_.html">✅ success</a></td></tr>
<tr><td>Stress test</td><td>Runs stateless functional tests concurrently from several clients to detect concurrency-related errors</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/bd9bfc34a932b732e08860abbe76dac10f9ee46b/stress_test__asan_.html">✅ success</a></td></tr>
<tr><td>Unit tests</td><td>Runs the unit tests for different release types</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/bd9bfc34a932b732e08860abbe76dac10f9ee46b/unit_tests__asan_.html">✅ success</a></td></tr>
<tr><td>Upgrade check</td><td>Runs stress tests on server version from last release and then tries to upgrade it to the version from the PR. It checks if the new server can successfully startup without any errors, crashes or sanitizer asserts</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/66908/07f642f1cac5253fda7eaa928ca5e659159b6b16/upgrade_check__debug_.html">✅ success</a></td></tr>
<tbody>
</table>
</details>

**Review by hanfei1991 — APPROVED** [review]

**Comment by zvonand** [comment]
@hanfei1991 let's merge this as is and see if it fixes the flakiness in master

**Comment by zvonand** [comment]
@hanfei1991 looks like it does not want to merge automatically

**Pull request merged**
